pkg/apis/beat/v1beta1/validations.go

```diff
@@ -24,13 +28,16 @@ var (
 		checkSingleConfigSource,
 		checkSpec,
 		checkAssociations,
+		checkMonitoring,
 	}
 
 	updateChecks = []func(old, curr *Beat) field.ErrorList{
 		checkNoDowngrade,
 	}
 
 	typeRegex = regexp.MustCompile("^[a-zA-Z0-9-]+$")
+	// minStackVersion is the minimum Stack version to enable Stack Monitoring on an Elastic Beats.
+	minStackVersion = version.MustParse("7.2.0-SNAPSHOT")
```

Comment:
As for the snapshot question, I was following the convention set out here https://github.com/elastic/cloud-on-k8s/commit/fb8205af22f9af286f1948e98678601a89fb8d83#diff-9f672c3af5fec0acba2a50cec9fa58743687d5f35d3f7e4c5591dcd63bca6dd4R27, and was assuming that this feature was enabled in the beats 7.2.0 snapshots prior to it's release.  The chances of someone running a 7.2.0 snapshot version at this stage is probably > .1%, but I was trying to be consistent with how we were handling this in the stackmon/validations package.

As for the internal monitoring being added in 7.2, I was looking at the [public monitoring documentation](https://www.elastic.co/guide/en/beats/filebeat/7.2/monitoring-internal-collection.html)
and this page doesn't exist prior to 7.2, but if you go back to 7.1 and look at the [configuration reference](https://www.elastic.co/guide/en/beats/filebeat/7.1/filebeat-reference-yml.html) you can see that monitoring existed, but the configuration was a different format:

```
# Uncomment to send the metrics to Elasticsearch. Most settings from the
# Elasticsearch output are accepted here as well. Any setting that is not set is
# automatically inherited from the Elasticsearch output configuration, so if you
# have the Elasticsearch output configured, you can simply uncomment the
# following line, and leave the rest commented out.
#xpack.monitoring.elasticsearch:
```

as opposed to 

```
monitoring:
  enabled: true
  elasticsearch:
```

So it technically existed prior to 7.2, but considering we'd have to support 2 different configuration formats, it seems sane to just support this >= 7.2.

Reply:
Ok makes sense. I think the snapshot versioning makes sense when we release a new feature at the same time as when the corresponding stack feature is released, so that we can test against pre-release versions.

Reply:
Removed `-SNAPSHOT`, and added a comment around why 7.2 is the min supported version.